x-pack/plugins/security/server/authentication/providers/kerberos.ts

```diff
@@ -267,17 +230,14 @@ export class KerberosAuthenticationProvider extends BaseAuthenticationProvider {
    * @param request Request instance.
    * @param [state] Optional state object associated with the provider.
    */
-  private async authenticateViaSPNEGO(
-    request: RequestWithLoginAttempt,
-    state?: ProviderState | null
-  ) {
-    this.debug('Trying to authenticate request via SPNEGO.');
+  private async authenticateViaSPNEGO(request: KibanaRequest, state?: ProviderState | null) {
+    this.logger.debug('Trying to authenticate request via SPNEGO.');
 
     // Try to authenticate current request with Elasticsearch to see whether it supports SPNEGO.
     let authenticationError: Error;
     try {
-      await this.options.client.callWithRequest(request, 'shield.authenticate');
-      this.debug('Request was not supposed to be authenticated, ignoring result.');
+      await this.getUser(request);
+      this.logger.debug('Request was not supposed to be authenticated, ignoring result.');
```

Comment:
I can, but only if we can be sure that `ElasticsearchErrorHelpers.isNotAuthorizedError` returns true for all 401 errors thrown from `elasticsearch-js` client and hence `callAsCurrentUser`.

IIRC when consumer passes `wrap401Errors = false` then 401 error will have a different type and `ElasticsearchErrorHelpers.isNotAuthorizedError` won't recognize it as a 401 error? Am I wrong?

Reply:
ok, let's use at least for challenges:
```ts
const challenges = ElasticsearchErrorHelpers.isNotAuthorizedError(authenticationError)
  ? ([] as string[]).concat(authenticationError.output.headers['WWW-Authenticate'])
  : []
```
otherwise we don't catch error in [tests](https://github.com/elastic/kibana/pull/39446/files#diff-5bea6b53b952964b695c319ed1ad4895R287) when Boom is removed from the core

Reply:
Sure, will update PR with these changes